build.gradle

```diff
@@ -18,9 +33,17 @@ plugins {
 
 allprojects {
     repositories {
-        google()
+        if (project.hasProperty("googleRepo")) {
```

Comment:
I wish we could factorize this logic in its own gradle function but sadly, build.gradle cannot anything but `buildscript` before `plugins`. Functions need to be define before they're called, as far as I see.

Reply:
I was chatting with @JohanLorenzo a bit about this, and the following are some notes on other potential solutions to this

---

The [plugin](https://docs.gradle.org/current/userguide/plugins.html) documentation lists a number of ways to specify plugins, some of which don't appear to have this restriction. My understanding is that the initial `buildscript` needs to be before plugins (hence can't use functions) to define the repo for plugins, however [this](https://docs.gradle.org/current/userguide/plugins.html#sec:custom_plugin_repositories) section from that page suggests that isn't used in any case.

Reply:
Chatted with @pocmo on Slack. 

sebastian  3 minutes ago
> Hmm, so we could try to add a local plugin that defines those functions?

> I assume that could work since google() is afaik also coming from the android plugin. But that would not work in the buildscript block. :man-shrugging:

jlorenzo  1 minute ago
> okay, I don't think it's worth making a plugin, then

jlorenzo  < 1 minute ago
> I'll merge it as it :slightly_smiling_face:

sebastian  < 1 minute ago
> :+1: